DNN Platform/Library/Services/GeneratedImage/ImageHandlerInternal.cs

```diff
@@ -231,6 +233,45 @@ public void HandleImageRequest(HttpContextBase context, Func<NameValueCollection
 
             string cacheId = GetUniqueIDString(context, uniqueIdStringSeed);
 
+            var hasprofileChanged = false;
+            var profilepic = context.Request.QueryString["mode"];
+            if (profilepic != null && profilepic == "profilepic")
+            {
+                var LogUserID = context.Request.QueryString["userId"];
+                if (!string.IsNullOrWhiteSpace(LogUserID))
+                {
+                    var LogPortalID = PortalSettings.Current.PortalId;
+                    int currentUserId;
+                    if (int.TryParse(LogUserID, out currentUserId))
+                    {
+                        var currentUser = UserController.GetUserById(LogPortalID, currentUserId);
+
+                        if (currentUser != null)
+                        {
+                            ProfileController.GetUserProfile(ref currentUser);
+
+                            //Get last update date for profile photo
+                            //21 is definiton Id for profile photo 
+                            var lastModifiedDateOfProfilePhoto = currentUser.Profile.ProfileProperties.GetById(21).LastModifiedDate;
```

Comment:
You are not supposed to use direct profile id value in the commits. This will ruin whole system if slipped into.